lib/ansible/modules/system/aix_filesystem.py

```diff
@@ -0,0 +1,577 @@
+        else:
+            changed = True
+            msg = rmfs_out
+            if not rmfs_out:
+                msg = "File system %s removed." % filesystem
+
+            return changed, msg
+    else:
+        changed = True
+        msg = ''
+
+        return changed, msg
+
+
+def mount_fs(module, filesystem):
+    """ Mount a file system. """
+    mount_cmd = module.get_bin_path('mount', True)
+
+    if not module.check_mode:
+        rc, mount_out, err = module.run_command(
+            "%s %s" % (mount_cmd, filesystem))
+        if rc != 0:
+            module.fail_json("Failed to run mount.", rc=rc, err=err)
+        else:
+            changed = True
+            msg = "File system %s mounted." % filesystem
+
+            return changed, msg
+    else:
+        changed = True
+        msg = ''
+
+        return changed, msg
+
+
+def unmount_fs(module, filesystem):
+    """ Unmount a file system."""
+    unmount_cmd = module.get_bin_path('unmount', True)
+
+    if not module.check_mode:
+        rc, unmount_out, err = module.run_command("%s %s" % (unmount_cmd, filesystem))
+        if rc != 0:
+            module.fail_json("Failed to run unmount.", rc=rc, err=err)
+        else:
+            changed = True
+            msg = "File system %s unmounted." % filesystem
+
+            return changed, msg
+    else:
+        changed = True
+        msg = ''
+
+        return changed, msg
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            account_subsystem=dict(type='bool', default=False),
+            attributes=dict(type='list', default=['agblksize="4096"', 'isnapshot="no"']),
+            auto_mount=dict(type='bool', default=True),
+            device=dict(type='str'),
+            filesystem=dict(type='str', required=True),
+            fs_type=dict(type='str', default='jfs2'),
+            permissions=dict(choices=['rw', 'ro'], default='rw'),
+            mount_group=dict(type='str'),
+            nfs_server=dict(type='str'),
+            rm_mount_point=dict(type='bool', default=False),
+            size=dict(type='str'),
+            state=dict(choices=['absent', 'present', 'mounted', 'unmounted'], default='present'),
+            vg=dict(type='str'),
+        ),
+        supports_check_mode=True,
+    )
+
+    account_subsystem = module.params['account_subsystem']
+    attributes = module.params['attributes']
+    auto_mount = module.params['auto_mount']
+    device = module.params['device']
+    fs_type = module.params['fs_type']
+    permissions = module.params['permissions']
+    mount_group = module.params['mount_group']
+    filesystem = module.params['filesystem']
+    nfs_server = module.params['nfs_server']
+    rm_mount_point = module.params['rm_mount_point']
+    size = module.params['size']
+    state = module.params['state']
+    vg = module.params['vg']
+
+    result = dict(
+        changed=False,
+        msg='',
+        state=state
```

Comment:
Is there any purpose in returning the state that the user already supplied ?
We try to avoid this generally.